Create rancher2_determine_leader.sh

oxr463 opened the pull request:
I noticed I was getting different results depending on how Rancher was deployed:

1. RKE cluster:

```sh
rancher-54d47dc9cf-prn4d is the leader in this Rancher instance
```

My RKE cluster is a single node running in AWS.

2. Local docker image:

```sh
4cfbfe99582c is the leader in this Rancher instance
```

The local docker image was spun up using this command:

```sh
sudo docker run -d --name rancher-2.5 --restart=unless-stopped -p 80:80 -p 443:443 --privileged rancher/rancher:v2.5.1
```

dnoland1 reviewed (commented):
You can get rid of the awks, trs, etc and just use jq:
```
kubectl -n kube-system get configmap cattle-controllers -o json | jq -r '.metadata.annotations."control-plane.alpha.kubernetes.io/leader"' | jq -r '.holderIdentity'
```

dnoland1 commented:
And might be nice to output this too:
```
 kubectl get pod -n cattle-system $LEADER -o custom-columns=NAME:.metadata.name,POD-IP:.status.podIP,HOST-IP:.status.hostIP
```

mattmattox reviewed (commented):
The output looks good. Could we add this to the [README](https://github.com/rancherlabs/support-tools/blob/master/troubleshooting-scripts/README.md)
We would just need the curl command and a sample output. 

```
mmattox@a1ubcubep01:~/RancherClusters/a1-k8s-prd$ kubectl -n kube-system get configmap cattle-controllers -o json | jq -r '.metadata.annotations."control-plane.alpha.kubernetes.io/leader"' | jq -r '.holderIdentity'
rancher-8cc7bf8d5-qklcc
mmattox@a1ubcubep01:~/RancherClusters/a1-k8s-prd$ kubectl get pod -n cattle-system $LEADER -o custom-columns=NAME:.metadata.name,POD-IP:.status.podIP,HOST-IP:.status.hostIP
NAME                               POD-IP        HOST-IP
rancher-8cc7bf8d5-nb9l2            10.42.4.127   172.27.3.33
rancher-8cc7bf8d5-qklcc            10.42.3.124   172.27.3.32
rancher-8cc7bf8d5-xq8kz            10.42.2.143   172.27.3.31
rancher-webhook-698dd9dc76-7djcw   10.42.4.126   172.27.3.33
mmattox@a1ubcubep01:~/RancherClusters/a1-k8s-prd$ 
````

dnoland1 reviewed (approved):
LGTM

dnoland1 commented:
Sorry, just tried the command in the README after merge and it doesn't work:
```
curl -s https://raw.githubusercontent.com/support-tools/troubleshooting-scripts/determine-leader/rancher2_determine_leader.sh | bash
bash: line 1: 404:: command not found
```
Could you make sure this is fixed?

dnoland1 commented on troubleshooting-scripts/determine-leader/rancher2_determine_leader.sh:
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+RANCHER_LEADER="$(kubectl -n kube-system get configmap cattle-controllers -o json | jq -r '.metadata.annotations."control-plane.alpha.kubernetes.io/leader"' | jq -r '.holderIdentity')"
+# Display Rancher Pods Information
+kubectl get pod -n cattle-system $LEADER -o custom-columns=NAME:.metadata.name,POD-IP:.status.podIP,HOST-IP:.status.hostIP
```
$LEADER should be $RANCHER_LEADER in this line.

oxr463 replied:
Should be fixed, check my rancher-leader branch for the changes.

oxr463 commented:
> Sorry, just tried the command in the README after merge and it doesn't work:
> 
> ```
> curl -s https://raw.githubusercontent.com/support-tools/troubleshooting-scripts/determine-leader/rancher2_determine_leader.sh | bash
> bash: line 1: 404:: command not found
> ```
> 
> Could you make sure this is fixed?

I copied the find-leader URL and modified it, so mine was wrong too. I updated both of them in the README.md file for my rancher-leader branch.

oxr463 commented:
See: https://github.com/rancherlabs/support-tools/pull/69
